File: OCaml Basics/25841.ml
open Scanf
open Printf
;;
let last n d =
    if n mod 10 = d then 1 else 0
;;
let rec count n d =
    if n<10 then last n d
    else last n d + count (n/10) d
;;
let rec sol l r d =
    if l = r then count l d
    else count l d + sol (l+1) r d
;;
scanf "%d %d %d\n" (fun l r d ->
    printf "%d\n" (sol l r d)
)
;;

File: C Basics/25841.c
# include <stdio.h>

int last(int n, int d){
    return n%10==d?1:0;
}

int count(int n, int d){
    return last(n,d)+(n<10?0:count(n/10,d));
}

int sol(int l, int r, int d){
    return count(l,d)+(l==r?0:sol(l+1,r,d));
}

int main(){
    int l,r,d;
    scanf("%d %d %d",&l,&r,&d);
    printf("%d",sol(l,r,d));
	return 0;
}
